### PR TITLE
[Improve][Flink-Kubernetes-V2] add exception throw

### LINKED ISSUE #2882: [Feature] [Flink-K8s-V2] Refactor the lifecycle control of Flink application-mode jobs on Kubernetes
It is a subtask of #2879 which includes the following content:
- Adapt [KubernetesNativeApplicationClient.doSubmit](https://github.com/apache/incubator-streampark/blob/dev/streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesNativeApplicationClient.scala#L38) to [FlinkOperator. deployApplicationJob](https://github.com/Al-assad/streampark-flink-kubernetes-v2/blob/master/src/main/scala/org/apache/streampark/flink/kubernetes/operator/FlinkK8sOperator.scala#L31);
- Adapt [KubernetesNativeApplicationClient.doCancel](https://github.com/apache/incubator-streampark/blob/dev/streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesNativeApplicationClient.scala#L91) to  [FlinkOperator.cancelJob，FlinkOperator.stopJob](https://github.com/Al-assad/streampark-flink-kubernetes-v2/blob/master/src/main/scala/org/apache/streampark/flink/kubernetes/operator/FlinkK8sOperator.scala#L51) ;
- Adapt [KubernetesNativeApplicationClient.doTriggerSavepoint](https://github.com/apache/incubator-streampark/blob/dev/streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesNativeApplicationClient.scala#L100) to [FlinkOperator.triggerJobSavepoint](https://github.com/Al-assad/streampark-flink-kubernetes-v2/blob/master/src/main/scala/org/apache/streampark/flink/kubernetes/operator/FlinkK8sOperator.scala#L56);
- Adapt KubernetesNativeApplicationClient.shutdown  to [FlinkK8sOperator.delete](https://github.com/Al-assad/streampark-flink-kubernetes-v2/blob/master/src/main/scala/org/apache/streampark/flink/kubernetes/operator/FlinkK8sOperator.scala#L41);
- Remove the step 4 to 8 in [FlinkK8sApplicationBuildPipeline](https://FlinkK8sApplicationBuildPipeline) ;
- Remove the remaining useless code.

Ref: [design doc](https://www.craft.me/s/LrTTzYHqSOLYqY)

### PR DESCRIPTION
**Pull request by caicancai**
In some places where it might be null, I add an exception throw
①[KubernetesApplicationClientV2.scala]
②[KubernetesSesionClientV2.scala]

## What changes were proposed in this pull request

fix: #2882 and #2884 

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / no)

### CODE REVIEW

**Comment by caicancai** [comment]
cc @Al-assad @wolfboys . It is true that the cluster part may throw an exception for null, but I'm not sure if there is a risk of application being null

**Comment by caicancai** [comment]
Whether it is necessary for DeployRequest to be as complete as SubRequest, it seems that DeployRequest is imperfect at present, and many configurable parameters have not been passed in

**Comment by caicancai** [comment]
For ② I think it is OK, ① I am not sure if it is appropriate to make such a change

**Comment by caicancai** [comment]
cc @Al-assad

File: streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesApplicationClientV2.scala [hunk 1]
```diff
@@ -103,12 +103,15 @@ object KubernetesApplicationClientV2 extends KubernetesClientV2Trait with Logger
...
 
-    val ingress = submitReq.k8sSubmitParam.ingressDefinition
+    val ingress = Option(submitReq.k8sSubmitParam.ingressDefinition)
+      .getOrElse(return Left("Ingress should not be empty"))
 
-    val imagePullPolicy = flinkConfObj
-      .getOption(KubernetesConfigOptions.CONTAINER_IMAGE_PULL_POLICY)
-      .map(_.toString)
-      .orElse(submitReq.k8sSubmitParam.imagePullPolicy)
+    val imagePullPolicy = Option(
```
Comment (Al-assad) [inline streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesApplicationClientV2.scala:109]: imagePullPolicy is a Option[String] here which is not mandatory to have a value.

File: streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesApplicationClientV2.scala [hunk 2]
```diff
@@ -103,12 +103,15 @@ object KubernetesApplicationClientV2 extends KubernetesClientV2Trait with Logger
       .filter(str => StringUtils.isNotBlank(str))
       .getOrElse(return Left("Flink base image should not be empty"))
 
-    val ingress = submitReq.k8sSubmitParam.ingressDefinition
+    val ingress = Option(submitReq.k8sSubmitParam.ingressDefinition)
```
Comment (Al-assad) [inline streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesApplicationClientV2.scala:106]: ingress should be a Option[IngressDef] here.

File: streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesApplicationClientV2.scala [hunk 3]
```diff
@@ -142,14 +145,17 @@ object KubernetesApplicationClientV2 extends KubernetesClientV2Trait with Logger
         .orElse(submitReq.k8sSubmitParam.jobManagerMemory)
         .getOrElse(KUBERNETES_JM_MEMORY_DEFAULT)
 
+      val ephemeralStorage = Option(submitReq.k8sSubmitParam.jobManagerEphemeralStorage)
```
Comment (Al-assad) [inline streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesApplicationClientV2.scala:148]: ephemeralStorage is is not a required parameter

**Review by Al-assad — CHANGES_REQUESTED** [review]
Thank you for your submission, but the code modified in the commit do not require a value. Therefore they are all of type Option in the original code.
I'm afraid I will reject this PR.

**Comment by caicancai** [comment]
> Thank you for your submission, but the code modified in the commit do not require a value. Therefore they are all of type Option in the original code. I'm afraid I will reject this PR.

Thank you for your reply and review . If the value is NULL, the return is null. There is no need to return an exception?

**Comment by caicancai** [comment]
I seem to understand what you mean. None doesn't affect the overall operation of the program, but Exception does, so those arguments can be None

**Comment by caicancai** [comment]
If so, I will shut this pr down

**Pull request closed without merging**